### PR TITLE
Issue #140 - HistoricalStream does not break _read_loop when on_data …

### PR DESCRIPTION
**Pull request by oughtotrade**
…returns False

### CODE REVIEW

File: betfairlightweight/streaming/betfairstream.py [hunk 1]
```diff
@@ -268,16 +268,28 @@ def __init__(self, directory, listener):
...
-        with open(self.directory, 'r') as f:
-            for update in f:
-                self.listener.on_data(update)
+        while self._running:
+            with open(self.directory, 'r') as f:
+                for update in f:
+                    if self.listener.on_data(update) is False:
+                        # if on_data returns an error stop the stream
+                        self.stop()
+                        raise ListenerError(self.listener.connection_id, update)
```
Comment (liampauling) [inline betfairlightweight/streaming/betfairstream.py]: No need to raise an error, just break out of the for loop and call stop()

File: betfairlightweight/streaming/betfairstream.py [hunk 2]
```diff
@@ -268,16 +268,28 @@ def __init__(self, directory, listener):
...
             self._read_loop()
 
+    def stop(self):
+        self._running = False
+
     def _read_loop(self):
-        with open(self.directory, 'r') as f:
-            for update in f:
-                self.listener.on_data(update)
+        while self._running:
```
Comment (liampauling) [inline betfairlightweight/streaming/betfairstream.py]: I think this needs to be moved into the for loop in some way as if you call stop() from outside the thread it won't actually stop the for loop

Comment (liampauling) [reply betfairlightweight/streaming/betfairstream.py]: Maybe a `if not self._running: break` after `self.listener.on_data(update)`

Comment (oughtotrade) [reply betfairlightweight/streaming/betfairstream.py]: OK!

**Pull request merged**